### PR TITLE
DefaultServlet - Strip BOM improvements.

### PR DESCRIPTION
**Pull request by jeanouii**
Draft PR mainly to gather feedback from community

### CODE REVIEW

File: java/org/apache/catalina/servlets/DefaultServlet.java [hunk 1]
```diff
@@ -335,8 +344,15 @@ public void init() throws ServletException {
             }
         }
 
-        if (getServletConfig().getInitParameter("useBomIfPresent") != null) {
-            useBomIfPresent = Boolean.parseBoolean(getServletConfig().getInitParameter("useBomIfPresent"));
+        final String useBomIfPresentConfig = getServletConfig().getInitParameter("useBomIfPresent");
+        if (useBomIfPresentConfig != null) {
+            if (!Arrays.asList("true", "false", "pass-through").contains(useBomIfPresentConfig)) {
```
[the remark this answers is not in the document]
Comment (rmaucher) [reply java/org/apache/catalina/servlets/DefaultServlet.java:349]: I think the important part in this sort of PR is the functionality and associated testing. There is always a round of polish afterwards these days.

[the remark this answers is not in the document]
Comment (rmaucher) [reply java/org/apache/catalina/servlets/DefaultServlet.java:349]: Good point on the compatibility, but the field is private right now in DefaultServlet, so it's ok to change to another type. The init-param config remains compatible.

[the remark this answers is not in the document]
Comment (jeanouii) [reply java/org/apache/catalina/servlets/DefaultServlet.java:349]: I'll get the changes in and see if I can add more tests.
Thanks for the quick follow up

[the remark this answers is not in the document]
Comment (jeanouii) [reply java/org/apache/catalina/servlets/DefaultServlet.java:349]: I made it such as it's fully backward compatible from a suer perspective and according to the discussions on the list. So we should be good in that regard.

@martin-g All good to me. All feedback always welcome

**Review by rmaucher — APPROVED** [review]

**Comment by markt-asf** [comment]
I have this applied locally along with an update to the documentation and the change log. I also refactored it to use an Enum for the configuration option.
I'll commit it shortly but I want to update the changelog processing first so we can link to PRs as well as BZ issues.

**Pull request closed without merging**